src/python/pants/backend/python/framework/django/dependency_inference.py

```diff
@@ -1,49 +1,122 @@
 # Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-from dataclasses import dataclass
+import json
+from pathlib import PurePath
 
 from pants.backend.python.dependency_inference.parse_python_dependencies import (
-    PythonDependencyVisitor,
-    PythonDependencyVisitorRequest,
-    get_scripts_digest,
+    ParsedPythonAssetPaths,
+    ParsedPythonDependencies,
+    ParsedPythonImportInfo,
+    ParsedPythonImports,
+)
+from pants.backend.python.dependency_inference.rules import (
+    ImportOwnerStatus,
+    PythonImportDependenciesInferenceFieldSet,
+    ResolvedParsedPythonDependencies,
+    ResolvedParsedPythonDependenciesRequest,
 )
 from pants.backend.python.framework.django.detect_apps import DjangoApps
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.target_types import EntryPoint
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.backend.python.util_rules.pex import PexRequest, VenvPex, VenvPexProcess
+from pants.core.util_rules.source_files import SourceFilesRequest
+from pants.core.util_rules.stripped_source_files import StrippedSourceFiles
 from pants.engine.fs import CreateDigest, FileContent
-from pants.engine.internals.native_engine import Digest, MergeDigests
+from pants.engine.internals.native_engine import Digest
 from pants.engine.internals.selectors import Get
+from pants.engine.process import ProcessResult
 from pants.engine.rules import collect_rules, rule
+from pants.engine.target import InferDependenciesRequest, InferredDependencies
 from pants.engine.unions import UnionRule
-from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+from pants.util.resources import read_resource
 
 
-@dataclass(frozen=True)
-class DjangoDependencyVisitorRequest(PythonDependencyVisitorRequest):
-    pass
+class InferDjangoDependencies(InferDependenciesRequest):
+    infer_from = PythonImportDependenciesInferenceFieldSet
 
 
-_scripts_package = "pants.backend.python.framework.django.scripts"
+_visitor_resource = "scripts/dependency_visitor.py"
 
 
 @rule
 async def django_parser_script(
-    _: DjangoDependencyVisitorRequest,
+    request: InferDjangoDependencies,
+    python_setup: PythonSetup,
     django_apps: DjangoApps,
-) -> PythonDependencyVisitor:
-    django_apps_digest = await Get(
-        Digest, CreateDigest([FileContent("apps.json", django_apps.to_json())])
+) -> InferredDependencies:
+    source_field = request.field_set.source
+    # NB: This doesn't consider https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-MIGRATION_MODULES
```

Comment:
This comment make me wonder if it wouldn't make sense to implement this as a hybrid Django-Pants plugin, to leverage the settings and Django-isms you get from a Django plugin, but run in the context of a Pants managed setting to avoid re-implementing/re-discovering how Django works under the covers.

Reply:
So far we've avoided Pants-specific plugins to frameworks.

(That's my whole comment. Really this PR is just to support ripping out the old stuff)

Reply:
> Really this PR is just to support ripping out the old stuff

yea, just got me thinking..

> So far we've avoided Pants-specific plugins to frameworks.

Actively avoided? (I've missed that memo 😅 ) ((if so, why?  OK, I can come up with a few cons to doing it that way... but also a few pros))

Reply:
I think it's fine to not worry about it. It's very rare to not call the migrations package `migrations`, and you always have the workaround of manual deps. If someone really needs this, they can add a Pants option to support that.